src/Form.js

```diff
@@ -0,0 +1,124 @@
+import {useState} from 'react';
+import Workout from './Workout';
+
+const Form = ({exercises, setExercises}) => {
+	const [userFilters, setUserFilters] = useState({
+		howMany: 0,
+		abs: false,
+		arms: false,
+		back: false,
+		chest: false,
+		legs: false,
+		shoulders: false,
+	});
+	const [filteredExercises, setFilteredExercises] = useState([]);
+
+	function handleChange(event) {
+		const {name, value, type, checked} = event.target;
+		setUserFilters((prev) => {
+			return {
+				...prev,
+				[name]: type === 'checkbox' ? checked : value,
+			};
+		});
+	}
+
+	const getWorkout = (event) => {
+		event.preventDefault();
+
+		let newArray = [];
+
+		if (userFilters.arms) {
+			newArray = exercises.filter((exercise) => exercise.category.id === 8);
+		}
+		if (userFilters.back) {
+			newArray = [...newArray, ...exercises.filter((exercise) => exercise.category.id === 12)];
+		}
+		if (userFilters.chest) {
+			newArray = [...newArray, ...exercises.filter((exercise) => exercise.category.id === 11)];
+		}
+		if (userFilters.core) {
+			newArray = [...newArray, ...exercises.filter((exercise) => exercise.category.id === 10)];
+		}
+		if (userFilters.legs) {
+			newArray = [...newArray, ...exercises.filter((exercise) => exercise.category.id === 9)];
+		}
+		if (userFilters.shoulders) {
+			newArray = [...newArray, ...exercises.filter((exercise) => exercise.category.id === 13)];
+		}
+
+		setFilteredExercises(randomize(newArray, userFilters.howMany));
+	};
+
+	const randomize = (array, numElements) => {
+		const shuffledArray = array.sort(() => 0.5 - Math.random());
+		const selectedElements = shuffledArray.slice(0, numElements);
+
+		return selectedElements;
+	};
+
+	const reset = (event) => {
+		event.preventDefault();
+		setUserFilters({
+			howMany: 0,
+			abs: false,
+			arms: false,
+			back: false,
+			chest: false,
+			legs: false,
+			shoulders: false,
+		});
+		setFilteredExercises([]);
+		console.log(userFilters);
+	};
+
+	console.log('form has rendered');
+
+	return (
+		<div className="wrapper">
+			<form onSubmit={getWorkout}>

```

Comment:
Is there a way to make this required? In the event that someone submits nothing, does something happen on the screen?